parser/dns.go

```diff
@@ -6,21 +6,37 @@ import (
 	"github.com/activecm/rita/parser/parsetypes"
 	"github.com/activecm/rita/pkg/data"
 	"github.com/activecm/rita/pkg/hostname"
-)
 
-func parseDNSEntry(parseDNS *parsetypes.DNS, filter filter, retVals ParseResults) {
+	log "github.com/sirupsen/logrus"
+)
 
-	// extract and store the dns client ip address
+func parseDNSEntry(parseDNS *parsetypes.DNS, filter filter, retVals ParseResults, logger *log.Logger) {
+	// get source destination pair
 	src := parseDNS.Source
-	srcIP := net.ParseIP(src)
 	dst := parseDNS.Destination
+
+	// parse addresses into binary format
+	srcIP := net.ParseIP(src)
 	dstIP := net.ParseIP(dst)
 
+	// verify that both addresses were able to be parsed successfully
+	if (srcIP == nil) || (dstIP == nil) {
+		logger.WithFields(log.Fields{
+			"uid": parseDNS.UID,
+			"src": parseDNS.Source,
+			"dst": parseDNS.Destination,
+		}).Error("Unable to parse valid ip address pair from http log entry, skipping entry.")
```

Comment:
This log message says http where it needs to say dns. I'll push up a quick fix and pull the branch in. Everything else looks good to me.